ibis/expr/types/relations.py

```diff
@@ -917,40 +917,173 @@ def aggregate(
 
     agg = aggregate
 
-    def distinct(self) -> Table:
-        """Compute the unique rows in `self`.
+    def distinct(
+        self,
+        *,
+        on: str | Iterable[str] | s.Selector | None = None,
+        keep: Literal["first", "last"] | None = "first",
+    ) -> Table:
+        """Return a Table with duplicate rows removed.
 
-        Returns
-        -------
-        Table
-            Unique rows of `self`
+        Similar to `pandas.DataFrame.drop_duplicates()`.
+
+        !!! note "Some backends do not support `keep='last'`"
```

Comment:
According to the tests, the backends won't work because they dont support window functions. That would imply that for these backends, it wouldn't work for any code path that goes down the group_by().agg() path, which would happen whenever on!=None. Am I understanding that correctly? So this note should be more like "Some backends do not support `on != None`"

Reply:
> According to the tests, the backends won't work because they dont support window functions.

The window function use in the test is unrelated to the functionality of distinct.

Reply:
It's used to ensure that the ordering of the output is comparable. This documentation line is correct and as intended.

Reply:
ahhh, I see, I thought the groupby caused a window function under the hood. Now I see it in the test.